core/src/main/java/com/google/instrumentation/trace/TraceExporter.java

```diff
@@ -54,6 +71,352 @@
+       */
+      public abstract Map<String, PerSpanNameSummary> getPerSpanNameSummary();
+
+      /** Summary of all available data for a span name. */
+      @AutoValue
+      @Immutable
+      public abstract static class PerSpanNameSummary {
+
+        PerSpanNameSummary() {}
+
+        /**
+         * Returns the number of active spans.
+         *
+         * @return the number of active spans.
+         */
+        public abstract int getNumberOfActiveSpans();
+
+        /**
+         * Returns the list of all latency based sampled buckets summary.
+         *
+         * @return the list of all latency based sampled buckets summary.
+         */
+        public abstract List<LatencyBucketSummary> getLatencyBucketSummaries();
+
+        /**
+         * Returns the list of all error based sampled buckets summary.
+         *
+         * @return the list of all error based sampled buckets summary.
+         */
+        public abstract List<ErrorBucketSummary> getErrorBucketSummaries();
+
+        /**
+         * Summary of a latency based sampled spans bucket. Contains {code Span} samples with
+         * latency between [latencyLowerNs, latencyUpperNs).
+         */
+        @AutoValue
+        @Immutable
+        public abstract static class LatencyBucketSummary {
+
+          LatencyBucketSummary() {}
+
+          /**
+           * Returns a new instance of {@code LatencyBucketSummary}.
+           *
+           * @param numSamples the number of sampled spans.
+           * @param latencyLowerNs the latency lower bound.
+           * @param latencyUpperNs the latency upper bound.
+           * @return a new instance of {@code LatencyBucketSummary}.
+           * @throws IllegalArgumentException if {@code numSamples} is negative.
+           */
+          public static LatencyBucketSummary create(
+              int numSamples, long latencyLowerNs, long latencyUpperNs) {
+            checkArgument(numSamples >= 0, "Negative numSamples.");
+            //CHECKSTYLE:OFF: Long class name.
+            return new AutoValue_TraceExporter_InProcessDebuggingHandler_Summary_PerSpanNameSummary_LatencyBucketSummary(
+                numSamples, latencyLowerNs, latencyUpperNs);
+            //CHECKSTYLE:ON: Long class name.
+          }
+
+          /**
+           * Returns the number of sampled spans in this bucket.
+           *
+           * @return the number of sampled spans in this bucket.
+           */
+          public abstract int getNumSamples();
+
+          /**
+           * Returns the latency lower bound of this bucket.
+           *
+           * @return the latency lower bound of this bucket.
+           */
+          public abstract long getLatencyLowerNs();
+
+          /**
+           * Returns the latency upper bound of this bucket.
+           *
+           * @return the latency upper bound of this bucket.
+           */
+          public abstract long getLatencyUpperNs();
```

Comment:
Same here, getLatencyLowerNs and getLatencyUpperNs could also specify inclusive/exclusive.